awslabs/aws-lambda-powertools-typescript
# Feat(logger): log event functionality

## Description of your changes

This PR introduces a new functionality: logging Lambda events. Developers can opt in for this behaviour by one of the following methods:
- passing the parameter `logEvent` set to `true` to the `injectLambdaContext` middleware
- passing the parameter `logEvent` set to `true` to the `injectLambdaContext` decorator
- setting the env var `POWERTOOLS_LOGGER_LOG_EVENT` to through while using the injectLambdaContext middleware/decorator

Note that this functionality is only available when using a middleware or decorator. 

The Lambda invocation will then print one log item with the following key-value pairs:
- `logLevel` INFO
- `message` key with value `Lambda invocation event`
- `event` key with the current Lambda invocation event as value





Logging a Lambda event is very typical use of a logger in AWS Lambda. This logic aims to simplify and reduce the code that developers need to maintain. 





### How to verify this change




See E2E tests results.

To see the docs:

```
npm run docs-website-build-run
```




![carbon (4)](https://user-images.githubusercontent.com/47529391/175299184-60cee8d2-daf4-464e-a3d9-8d54a7059653.png)

### Related issues, RFCs


[#482](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/482)  

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist


### Breaking change checklist


---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
